pkg/operator/deploy/deploy.go

```diff
@@ -196,6 +196,33 @@ func (o *operator) resources(ctx context.Context) ([]kruntime.Object, error) {
 	}
 
 	// then dynamic resources
+	if o.oc.UsesWorkloadIdentity() {
+		var operatorIdentity *api.PlatformWorkloadIdentity // use a pointer to make it easy to check if we found an identity below
+		for _, i := range o.oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+			if i.OperatorName == pkgoperator.OperatorIdentityName {
+				operatorIdentity = &i
+			}
+		}
+
+		if operatorIdentity == nil {
+			return nil, errors.New(fmt.Sprintf("operator identity %s not found", pkgoperator.OperatorIdentityName))
+		}
+
+		results = append(results, &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: pkgoperator.Namespace,
+				Name:      pkgoperator.OperatorIdentitySecretName,
+			},
+			StringData: map[string]string{
+				"azure_client_id":            operatorIdentity.ClientID,
+				"azure_tenant_id":            o.env.TenantID(),
+				"azure_region":               o.oc.Location,
+				"azure_subscription_id":      o.env.SubscriptionID(),
```

Comment:
If you get the tenant ID and sub ID from `o.env`, that data is coming from the RP's instance metadata and isn't correct for the platform managed identities. To get the correct values for the ARO operator's managed identity, you should parse the MI's resource ID using [`arm.ParseResourceID`](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/arm#ParseResourceID).

Reply:
The resource ID contains the subscription ID but not the tenant ID. For the tenant ID, do we need to pull that from the subscription doc, or does it make sense to use `o.oc.Identity.TenantID` (which is populated by MSI-RP)?

It seems like we should use the subscription doc.

Reply:
Yeah, Alex and I discussed this in Slack and he brought it to my attention that the resource ID doesn't include the tenant ID. I recommended using the Azure SDK to `GET` the MI by resource ID and then pull the tenant ID from the response, but I guess using the subscription doc works too since we're assuming that all MIs are in the same tenant as the ARO cluster sub.

Reply:
Thanks for the explanations! I've updated this to use the subscription doc instead of `o.env`

Reply:
I should've mentioned this earlier, and maybe I'm being overly cautious, but I would still take the subscription ID from the MI's resource ID to be super safe. The MI might be in the same tenant but a different sub, and I don't see why we couldn't support that.

Reply:
Pointing more around how the workload identity client is created, the subscriptionID is not needed for creating the client. Ref:- https://azure.github.io/azure-workload-identity/docs/quick-start.html
SubscriptionID is provided as part of secret to point out the scope where those Managed Identities will be used and not where that MI reside, this is similar to the region.
So, I believe there isn't a need to take the subscriptionID from the MI and rather use the subscriptionID from the cluster doc.
PS- Trying to find better source to point that out.

Reply:
It is similar to the usage here:- https://github.com/openshift/machine-api-provider-azure/blob/main/pkg/cloud/azure/actuators/machine_scope.go#L454

Reply:
Oh, I see. Thank you for pointing that out. Then I agree that the cluster doc is the best place to get the subscription ID from.

Reply:
Thanks for the discussion!

Is the subscription doc is accessible from [an operator variable|https://github.com/Azure/ARO-RP/blob/a1c4de75a48b/pkg/operator/deploy/deploy.go#L65-L76]? I'm not seeing an obvious path, and and the operator variable is all this function currently has access to

If there isn't a chain of fields to use to get the subscription doc, would using [arm.ParseResourceID()|https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/arm#ParseResourceID] with `operator.oc.ID` be acceptable?

Reply:
Apologies, I don't get the question exactly.
IMO, the existing way in PR to pass the subscription doc and getting the subscriptionID is correctly done and there's no change needed there.
@kimorris27 please confirm the same.

Reply:
>If there isn't a chain of fields to use to get the subscription doc, would using [arm.ParseResourceID()](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/arm#ParseResourceID) with operator.oc.ID be acceptable?

This would technically work too, but I agree with Rajdeep that the way you have it right now looks good. The work to add the subscription doc as a field on the `operator` type is already done.

Reply:
Sorry, I meant cluster doc, not subscription doc in that comment

Regardless, it sounds like things are good now, though